src/modules/flyout/flyout.component.ts

```diff
@@ -1,32 +1,41 @@
 import {
-  AnimationTransitionEvent,
+  Component,
   ChangeDetectionStrategy,
   ChangeDetectorRef,
-  Component,
-  ViewChild,
   ComponentFactoryResolver,
-  EventEmitter,
+  ElementRef,
   Injector,
+  OnDestroy,
+  OnInit,
   ReflectiveInjector,
+  Type,
+  ViewChild,
   ViewContainerRef
 } from '@angular/core';
+
 import {
   animate,
+  AnimationEvent,
   state,
   style,
   transition,
   trigger
 } from '@angular/animations';
 
+import { Subject } from 'rxjs/Subject';
+import 'rxjs/add/operator/takeUntil';
+
+import { SkyFlyoutAdapterService } from './flyout-adapter.service';
+import { SkyFlyoutInstance } from './flyout-instance';
+
 import {
   SkyFlyoutConfig,
-  SkyFlyoutInstance
-} from './index';
-import { SkyFlyoutAdapterService } from './flyout-adapter.service';
-import { SkyFlyoutService } from './flyout.service';
+  SkyFlyoutMessage,
+  SkyFlyoutMessageType
+} from './types';
 
-const FLYOUT_OPEN_STATE: string = 'flyoutOpen';
```

Comment:
Just curious - why are we removing the type declaration here?

Reply:
Sure thing. This is one of the "pedantic style rules" we've adopted for some of our newer components. The type is inferred when being set by a deliberate value. It's purely preference to do it this way, but I find that it saves the code from over-complication since the right-hand assignment will always declare the property's type. For example,

```
class Foo {
  public bar: string = 'baz'; // of course it's a string :-P
  public baz: number = 'foo'; // makes no sense
}
```

vs.

```
class Foo {
  public bar = 'baz'; // simple and to the point
}
```

Here's some more background: https://stackoverflow.com/a/45911822/6178885, and https://angular-2-training-book.rangle.io/handout/features/type_inference.html.